doc/specs/#5000 - Process Model 2.0/#5000 - Process Model 2.0.md

```diff
@@ -0,0 +1,881 @@
+
+11. (Dependent on 9, 10) `wt.exe` needs to be able to spawn as a content
+    process, accepting a GUID for its ID, and spawning a single control core.
+   - When the content process is first spawned, it won't create the core or
+     connection, nor will it have any settings. The first client to connect to
+     the content process should make sure to set up the settings before
+     initializing the control.
+   - A scratch XAML Island application might be a useful tool at this point, to
+     test hosting the content in another process (that's not a full-blown
+     terminal instance).
+
+12. (Dependent on 11) TerminalApp creates new content processes for each and
+    every terminal instance it spawns. At this point, there's no tear-out, but
+    the terminal instances are all out-of-proc from the window process.
+
+13. (Dependent on 12) Terminal can drop tabs onto another WT window process by
+    communicating the structure of the tab's panes and their content GUIDs.
+   - At this point, the tabs can't be torn out to create new windows, only move
+     between existing windows
+   - It might be hard to have the tab "attach" to the tab row of the other window.
+   - It might be easiest to do this after 1, and communicate to the new window
+     process the GUID of the old window process and the tab within the original
+     window process, and just have the new window process ask the old window
+     process what the structure of the tab is
+     - Though, the tab won't be in the original process's list of tabs anymore,
+       so that might not be helpful.
+
+14. `wt` accepts an initial position, size on the commandline.
+
+15. (Dependent on 13, 14) Tearing out a tab and dropping it _not_ on another WT
+    window creates a new window for the tab.
+   - This will require and additional argument to `wt` for it to be able to
+     inherit the structure of a given set of tabs. Either this will need to be
+     passed on the cmdline, or the newly spawned process will need to be able to
+     communicate with the original process to ask it what structure it should be
+     building.
+   - Doing this after 1 might be helpful.
+   - Needs 14 to be able to specify the location of the new window.
+
+<hr>
+
+16. Add support for a `NewWindow` action
+17. `wt` accepts a commandline arg to force it to auto-elevate itself if it
+    isn't already elevated.
+18. (Dependent on 16, 17) Users can mark a profile as `"elevated": true`, to
+    always force a new elevated window to be created for elevated profiles.
+19. (Dependent on 18, 13) When a user creates an elevated profile, we'll
+    automatically create an elevated version of the current window, and move all
+    the current tabs to that new window.
+
+## Footnotes
+
+<a name="footnote-1"><a>[1]: See [Quake mode scenarios] for a longer enumeration
+of possible scenarios.
+
+## Future considerations
+
+* Yea add these
+
+* Non-terminal content in the Terminal. We've now created a _very_

```

Comment:
Since Settings UI is in scope for WT v2.0, shouldn't this be moved from "Future Considerations" to "Potential Issues"? If there's a specific change you have in mind for a tab to host a Settings UI, I guess it would make more sense to do that now, right?

Reply:
I've added a super long section on how this will work with the settings UI, because I forgot that we're putting it in a tab in 2.0 🤦‍♂️